src/Microsoft.Extensions.DependencyInjection/ServiceLookup/ClosedIEnumerableService.cs

```diff
@@ -3,19 +3,18 @@
 
 using System;
 using System.Collections.Generic;
-using System.Linq;
 
 namespace Microsoft.Extensions.DependencyInjection.ServiceLookup
 {
     internal class ClosedIEnumerableService : IService
     {
-        private readonly Type _itemType;
         private readonly ServiceEntry _serviceEntry;
 
         public ClosedIEnumerableService(Type itemType, ServiceEntry entry)
         {
-            _itemType = itemType;
+            ServiceType = itemType;
             _serviceEntry = entry;
+            ImplementationType = itemType.MakeArrayType();
```

Comment:
If this data is used only for expression tree  generation can we just make a function in `CallSiteExpressionBuilder` that does this?